src/libs/ReactNativeLauncher.js

```diff
@@ -244,6 +255,116 @@ class ReactNativeLauncher extends Launcher {
+    }
+    try {
+      log.info(
+        `getCookie on this domain : ${cookieDomain} for account ${this.startContext.account.id}`
+      )
+      const cookies = await CookieManager.get(cookieDomain)
+      log.info('CookieManager.get =>', cookies)
+      return cookies
+    } catch (err) {
+      throw new Error(`Error in worker: ${err.message}`)
+    }
+  }
+
+  /**
+   * This method returns one cookie matching the provided name when asked by the running webview(s).
+   * This method is callable by the content script.
+   *
+   * @param {String} options.cookieName : wanted cookie's by its name.
+   */
+  async getCookieFromKeychain(cookieName) {
+    log.info('Starting getCookieFromKeychain in RNLauncher')
+    try {
+      const { account, manifest } = this.startContext
+      const accountId = account.id
+      const konnectorSlug = manifest.slug
+      log.info('Checking if cookie is present with this name :', cookieName)
+      const existingCookies = await getCookie({
+        accountId,
+        konnectorSlug,
+        cookieName
+      })
+      if (existingCookies !== null) {
+        return existingCookies
+      }
+      log.info(`No cookie named "${cookieName}" has been found, retunring null`)
+      return null
+    } catch (err) {
+      throw new Error(
+        `Error in worker during getCookieFromKeychain: ${err.message}`
+      )
+    }
+  }
+
+  /**
+   * This method saves given cookie in the keychain when asked by the running webview(s).
+   * This method is callable by the content script.
+   *
+   * @param {cookieObject.<Object>} : Object containing all the needed properties of the cookie.
+   */
+  async saveCookie(cookieObject) {
+    log.info('Starting saveCookie in RNLauncher')
+    try {
+      const { account, manifest } = this.startContext
+      const accountId = account.id
+      const konnectorSlug = manifest.slug
+      log.info(
+        'Checking if cookie is present with this payload :',
+        cookieObject
+      )
+      const existingCookies = await getCookie({
+        accountId,
+        konnectorSlug,
+        cookieObject
+      })
+      log.info('Cookie from keychain', existingCookies)
+      log.info('Cookie Object', { accountId, konnectorSlug, cookieObject })
+      if (existingCookies !== null) {
+        log.info('gettin existing cookie condition')
+        await removeCookie(accountId, konnectorSlug)
+      }
+      await saveCookie({ accountId, konnectorSlug, cookieObject })
+    } catch (err) {
+      throw new Error(`Error in worker during saveCookie: ${err.message}`)
+    }
+  }
+
+  /**
+   * This method returns a webView Cookie find by its name when asked by the running webview(s).
+   * This method is callable by the content script.
+   *
+   * @param {String} cookieDomain : Domain to recover cookies from .
+   * @param {String} cookieName : Name of the wanted cookie.
+   */
+  async getWebViewCookie(cookieDomain, cookieName) {
```

Comment:
So we have 3methods ! 
```
getCookies(cookieDomain)
getCookieFromKeychain(cookieName)
getWebViewCookie(cookieDomain, cookieName)
```

Do we really need `getCookieFromKeychain(cookieName)`? Shouldn't we always use `getWebViewCookie(cookieDomain, cookieName)`?

cc @Ldoppea

Reply:
There is a naming problem here as `getCookies()` is retrieving cookies from the WebView. We should have
- `getCookieFromKeychain()` => rename to `getCookies()` and return ALL cookies from keychain
- `getCookies()` => rename to `getWebViewCookies()`
- `getWebViewCookie()` => no change

We initially expected to have only `getCookies(domain)` and `getWebViewCookie(domain, name)`

`getWebViewCookie(domain, name)` should be used when the cookie is HttpOnly. But there is some scenario where we don't know the cookie name. For example `sess-cozySOMERANDOMID`. So we chose to add a `getWebViewCookies(domain)` (but we didn't update the paper sry).